atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/atomic/AtomicTable.java

```diff
@@ -25,25 +25,26 @@
 import java.util.Map;
 
 /**
- * A generally persisted key value store that supports an atomic put unless exists operation.
+ * A generally persisted key value store that supports an atomic update operation.
  * @param <K> Key for the mapping
  * @param <V> Value for the mapping
  */
-public interface PutUnlessExistsTable<K, V> {
+public interface AtomicTable<K, V> {
+
     /**
-     * Atomic put unless exists. If the method does not throw, any subsequent get is guaranteed to return V. If the
+     * Atomic update. If the method does not throw, any subsequent get is guaranteed to return V. If the
      * method throws an exception, subsequent gets may return either V or null but once V is returned subsequent calls
      * are guaranteed to return V.
```

Comment:
Is this still correct? In a sense, if this is a PUE/CAS backed store, subsequent gets may return V _or the previous value_, but once V is returned subsequent calls are guaranteed to return V